github_app.go

```diff
@@ -96,142 +196,70 @@ func getInstallationID(ctx context.Context, owner, repo string) (int64, error) {
 	return installation.InstallationID, nil
 }
 
-type githubAppAuth struct {
-	appId      int64
-	privateKey []byte
-}
-
-// getGithubAppAuth returns the app id and app private key if they exist.
-func (s *Settings) getGithubAppAuth() *githubAppAuth {
-	if s.AuthConfig.Github == nil || s.AuthConfig.Github.AppId == 0 {
-		return nil
-	}
-
-	key := s.Expansions[githubAppPrivateKey]
-	if key == "" {
-		return nil
-	}
-
-	return &githubAppAuth{
-		appId:      s.AuthConfig.Github.AppId,
-		privateKey: []byte(key),
-	}
-}
-
-// CreateInstallationToken uses the owner/repo information to request an github app installation id
-// and uses that id to create an installation token.
-func (s *Settings) CreateInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
-	const (
-		maxDelay   = 10 * time.Second
-		minDelay   = time.Second
-		maxRetries = 5
-	)
-
-	if owner == "" || repo == "" {
-		return "", errors.New("no owner/repo specified to create installation token")
-	}
-	authFields := s.getGithubAppAuth()
-	if authFields == nil {
-		return "", errors.New("GitHub app is not configured in admin settings")
-	}
-
+func getGitHubClientForAuth(authFields *githubAppAuth) (*github.Client, error) {
 	retryConf := utility.NewDefaultHTTPRetryConf()
-	retryConf.MaxDelay = maxDelay
-	retryConf.BaseDelay = minDelay
-	retryConf.MaxRetries = maxRetries
+	retryConf.MaxDelay = GitHubRetryMaxDelay
+	retryConf.BaseDelay = GitHubRetryMinDelay
+	retryConf.MaxRetries = GitHubMaxRetries
 
 	httpClient := utility.GetHTTPRetryableClient(retryConf)
-	defer utility.PutHTTPClient(httpClient)
 
 	key, err := jwt.ParseRSAPrivateKeyFromPEM(authFields.privateKey)
 	if err != nil {
-		return "", errors.Wrap(err, "parsing private key")
+		return nil, errors.Wrap(err, "parsing private key")
 	}
 
 	itr := ghinstallation.NewAppsTransportFromPrivateKey(httpClient.Transport, authFields.appId, key)
 	httpClient.Transport = itr
 	client := github.NewClient(httpClient)
+	return client, nil
+}
 
-	installationID, err := getInstallationID(ctx, owner, repo)
+func getInstallationIDFromGitHub(ctx context.Context, authFields *githubAppAuth, owner, repo string) (int64, error) {
```

Comment:
I would suggest adding a comment to this function or file explaining that we have GitHub calls outside of thirdparty because they're used for initiatlizing the environment, so we don't try to fix the import cycle again alter